tools/inference_engine_tests/test_cases/test_model.cpp

```diff
@@ -128,5 +128,73 @@ TEST(ModelTest, MatMul_fused_activation)
     destroy_node_if_valid(input_a);
     destroy_node_if_valid(input_b);
     destroy_node_if_valid(port_out);
+    destroy_node_if_valid(port_matmul_out);
     inferenceEngineDestroyContext(ctx);
 }
+
+TEST(ModelTest, MatMul_6_nodes)
+{
+    // *   *  *
+    //  \ /  /
+    //   *  *  // matmul, activation
+    //    \/
+    //     * // mat mul
+
+    auto device = reinterpret_cast<inference_engine_device_t>(G_DX12_ENGINE.d3d12_device.Get());
+    auto stream = reinterpret_cast<inference_engine_stream_t>(G_DX12_ENGINE.command_list.Get());
+    auto ctx = inferenceEngineCreateContext(device, fill_with_dx12_callbacks());
+
+    inference_engine_port_desc_t input_desc{};
+    input_desc.tensor.data_type = inference_engine_data_type_t::INFERENCE_ENGINE_DATA_TYPE_FP16;
+    set_array(input_desc.tensor.dims, 1, 1, 16, 16);
+    // create 2 port with the same description
+    // M = 32, K = 16, N = 16
+    auto input_a = inferenceEngineCreatePort(input_desc);
+    EXPECT_TRUE(input_a != nullptr);
+    auto input_b = inferenceEngineCreatePort(input_desc);
+    EXPECT_TRUE(input_b != nullptr);
+    auto input_c = inferenceEngineCreatePort(input_desc);
+    EXPECT_TRUE(input_c != nullptr);
+
+    // MatMul
+    inference_engine_matmul_desc_t matmul_desc{};
+    matmul_desc.input_a = input_a;
+    matmul_desc.input_b = input_b;
+    auto port_matmul_out = inferenceEngineCreateMatMul(matmul_desc);
+    EXPECT_TRUE(port_matmul_out != nullptr);
```

Comment:
Naming of the ports could be a bit different in this test
`port_matmul_out ` -> `port_matmul_a` 
`port_out ` -> `port_activation`